noir-projects/aztec-nr/aztec/src/macros/mod.nr

```diff
@@ -235,59 +274,63 @@ comptime fn generate_process_log() -> Quoted {
                         f"Expected packed note of length {expected_len} but got {actual_len} for note type id {note_type_id}"
                     );
 
-                    aztec::note::utils::compute_note_hash_and_optionally_a_nullifier($typ::unpack, contract_address, nonce, true, storage_slot, packed_note.storage())
+                    aztec::note::utils::compute_note_hash_and_optionally_a_nullifier($unpack, contract_address, nonce, true, storage_slot, packed_note.storage())
                 }
             },
         );
     }
 
     let if_note_type_id_match_statements = if_note_type_id_match_statements_list.join(quote {});
 
-    let body = if notes.len() > 0 {
+    if notes.len() > 0 {
         quote {
-            // Because this unconstrained function is injected after the contract is processed by the macros, it'll not
```

Comment:
Had to do this change because if we returned a body from the if and then return the final quote from the function later we would get warnings in cases where there are no notes as the function args would be unused.